servicebus/data-plane/azure-servicebus/src/main/java/com/microsoft/azure/servicebus/MessageReceiver.java

```diff
@@ -762,17 +763,9 @@ public IMessage peek(long fromSequenceNumber) throws InterruptedException, Servi
 
     private void schedulePruningRequestResponseLockTokens() {
         // Run it every 1 hour
-        Timer.schedule(new Runnable() {
-            public void run() {
-                Instant systemTime = Instant.now();
-                Entry<UUID, Instant>[] copyOfEntries = (Entry<UUID, Instant>[]) MessageReceiver.this.requestResponseLockTokensToLockTimesMap.entrySet().toArray();
-                for (Entry<UUID, Instant> entry : copyOfEntries) {
-                    if (entry.getValue().isBefore(systemTime)) {
-                        // lock expired
-                        MessageReceiver.this.requestResponseLockTokensToLockTimesMap.remove(entry.getKey());
-                    }
-                }
-            }
+        Timer.schedule(() -> {
+            Instant systemTime = Instant.now();
+            MessageReceiver.this.requestResponseLockTokensToLockTimesMap.entrySet().removeIf(entry -> entry.getValue().isBefore(systemTime));
         }, Duration.ofSeconds(3600), TimerType.RepeatRun);
```

Comment:
[SpotBugs P1] Impossible downcast of toArray() result: : The toArray() of almost all collections return an Object[]. They can't really do anything else, since the Collection object has no reference to the declared generic type of the collection.